pkg/controller/plan/migration.go

```diff
@@ -916,7 +943,124 @@ func (r *Migration) resetPrecopyTasks(vm *plan.VMStatus, step *plan.Step) {
 	}
 }
 
-//
+func (r *Migration) createVolumes(vm ref.Ref) (err error) {
+	if r.vmMap == nil {
+		r.vmMap, err = r.kubevirt.VirtualMachineMap()
+		if err != nil {
+			return
+		}
+	}
+	ovirtVm := &ovirt.Workload{}
+	err = r.Source.Inventory.Find(ovirtVm, vm)
+	if err != nil {
+		return
+	}
+	url, err := url.Parse(r.Source.Provider.Spec.URL)
+	if err != nil {
+		return
+	}
+
+	storageName := &r.Context.Map.Storage.Spec.Map[0].Destination.StorageClass
+	for _, da := range ovirtVm.DiskAttachments {
+		populatorCr := v1beta1.OvirtImageIOPopulator{
+			ObjectMeta: meta.ObjectMeta{
+				Name:      da.DiskAttachment.ID,
+				Namespace: r.Plan.Spec.TargetNamespace,
+			},
+			Spec: v1beta1.OvirtImageIOPopulatorSpec{
+				EngineURL:        fmt.Sprintf("https://%s", url.Host),
+				EngineSecretName: r.Source.Secret.Name,
+				DiskID:           da.Disk.ID,
+			},
+		}
+
+		err = r.Client.Create(context.Background(), &populatorCr, &client.CreateOptions{})
+		if err != nil {
+			return
+		}
+		apiGroup := "forklift.konveyor.io"
```

Comment:
we can take it from [here](https://github.com/kubev2v/forklift/blob/main/pkg/apis/forklift/v1beta1/register.go#L33)